### ISSUE TITLE
Destroy or delete method

### ISSUE THREAD

**Issue by marksiemers** [mosop/teeplate#8]
Would it be possible to provide a `destroy` or `delete` method that is the opposite of `render` in that it will remove any files that would have been created by `render`?

**Comment by aaronstillwell** [comment]
I'd be happy to take this on and file a PR in the near future. (I've come here from https://github.com/amberframework/amber/issues/188 as above).

I've reviewed some of the source and I've skimmed over the original issue for appending to files   (https://github.com/mosop/teeplate/issues/2). Is it logical to assume that using a `-` prefix might be suitable for marking a file for destruction?

Happy to hear some feedback on this before filing a PR @mosop

**Comment by marksiemers** [comment]
@aaronstillwell - Thanks for offering to take this one.

As far as having a `-` indicate that a file would be deleted. That wouldn't match my use-case.

What I was looking for is something that takes the exact same logic to decide what files are `render`ed.

It could be the same structure as [this method](https://github.com/mosop/teeplate/blob/master/src/lib/renderer.cr#L86):
```crystal
def render
  begin
    @entries.each do |entry|
      if @per_entry
        entry.render
      else
        entry.confirm
      end
    end
    @entries.each(&.render) unless @per_entry
  rescue ex : Quit
    @quitted = true
  end
end
```

Just, instead of `entry.render`, it would be `entry.delete` and instead of `entry.confirm`, `entry.confirm.delete`

If the minus (`-`) designation were to be used, I would expect it to remove lines from the file that match.

**Comment by aaronstillwell** [comment]
Woops, gotcha @marksiemers, you're absolutely right the `-` makes no sense. Happy to take this on using the approach you've described though :ok_hand:

**Comment by marksiemers** [comment]
@aaronstillwell - Do you think you'll have time to work through this in the next week or so?

The amber team is updating some milestones and we're looking where to bucket a change that depends on this functionality.

If you're PR doesn't get approved here, we may do a fork into the amberframework org, and we could incorporate it there.

**Comment by aaronstillwell** [comment]
@marksiemers yes I plan to have it done this weekend. I'll happily move the resulting PR to the fork if it comes to that.

**Comment by marksiemers** [comment]
@aaronstillwell - I screwed up the forking on this, but it should still be fine: https://github.com/amberframework/teeplate

You can open a PR there.
